packages/sources/coinmetrics/src/endpoint/burned.ts

```diff
@@ -1,63 +1,164 @@
+    burnedETH = burnedETH.plus(feeTotNTV.minus(revNtv.minus(issTotNtv)))
+  })
+  return burnedETH
 }
 
 export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
   const validator = new Validator(request, inputParameters)
   if (validator.error) throw validator.error
 
+  const today = new Date().toISOString().split('T')[0]
+
   const jobRunID = validator.validated.id
-  const asset = validator.overrideSymbol(AdapterName, validator.validated.data.asset)
-  const frequency = validator.validated.data.frequency || DEFAULT_FREQUENCY
-  const pageSize = validator.validated.data.pageSize || DEFAULT_PAGE_SIZE
-  const url = 'timeseries/asset-metrics'
-  const metrics = 'FeeTotNtv,RevNtv,IssTotNtv'
+  const startDate = validator.validated.data.startDate || DEFAULT_START_DATE
+  const endDate = validator.validated.data.endDate || today
+
+  if (startDate && !reDate.test(startDate)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'startDate': ${startDate}. Expected formats is: YYYY-MM-DD.`,
+      statusCode: 400,
+    })
+  }
+
+  if (Date.parse(startDate) > Date.parse(today)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'startDate': ${startDate}. Is greater than today: ${today}.`,
+      statusCode: 400,
+    })
+  }
+
+  if (!reDate.test(endDate)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'endDate': ${endDate}. Expected formats is: YYYY-MM-DD.`,
+      statusCode: 400,
+    })
+  }
+
+  if (Date.parse(endDate) > Date.parse(today)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid 'endDate': ${endDate}. Is greater than today: ${today}.`,
+      statusCode: 400,
+    })
+  }
+
+  if (Date.parse(startDate) > Date.parse(endDate)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Invalid date range: 'startDate': ${startDate} cannot be greater than 'endDate': ${endDate}.`,
+      statusCode: 400,
+    })
+  }
 
   const params = {
-    assets: (asset as string).toLowerCase(),
-    metrics,
-    frequency,
+    assets: ASSETS,
+    metrics: METRICS,
+    frequency: FREQUENCY,
+    page_size: PAGE_SIZE,
+    start_time: startDate,
+    end_time: endDate,
     api_key: config.apiKey,
-    page_size: pageSize,
   }
 
-  const options = { ...config.api, params, url }
+  const options = { ...config.api, params, url: URL }
+
+  let totalBurnedETH = new Decimal(0)
+  let response
+  /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
+  while (true) {
+    response = await Requester.request(options)
 
-  const response = await Requester.request<ResponseSchema>(options)
-  const FeeTotNtv = ethers.utils.parseEther(response.data.data[0].FeeTotNtv)
-  const RevNtv = ethers.utils.parseEther(response.data.data[0].RevNtv)
-  const IssTotNtv = ethers.utils.parseEther(response.data.data[0].IssTotNtv)
-  const result = FeeTotNtv.sub(RevNtv.sub(IssTotNtv))
+    const responseData: ResponseData = response.data
```

Comment:
This is no longer necessary with the change above

```suggestion
    const responseData = response.data
```